### PR TITLE
[#2968] Rails 4.0

### PR DESCRIPTION
**Pull request by garethrees**
Work on #2968 

Currently doing the minimum to get to Rails 4.0.

### CODE REVIEW

**Comment by garethrees** [comment]
Current failures for reference:

```
  1) PublicBodyHelper#type_of_authority when in a non-default locale creates the anchor href in the correct locale
     Failure/Error:
       I18n.with_locale(:es) { expect(type_of_authority(public_body)
                                     ).to eq(anchor) }

       expected: "<a href=\"/es/body/list/spec\">Spec category</a>"
            got: "<a href=\"/body/list/spec\">Spec category</a>"

       (compared using ==)
     # ./spec/helpers/public_body_helper_spec.rb:133:in `block (5 levels) in <top (required)>'
     # /home/vagrant/bundle/ruby/1.9.1/gems/i18n-0.7.0/lib/i18n.rb:257:in `with_locale'
     # ./spec/helpers/public_body_helper_spec.rb:133:in `block (4 levels) in <top (required)>'

  2) GeneralController when using xapian search when passed a non-HTML request treats invalid formats as html
     Failure/Error: expect(response.status).to eq(200)

       expected: 200
            got: 404

       (compared using ==)
     # ./spec/controllers/general_controller_spec.rb:404:in `block (3 levels) in <top (required)>'

  3) when deriving a name, email and formatted address from a message from a line should quote a name with quotes in it
     Failure/Error: expect([name, email, address]).to eq(expected_result)

       expected: ["FOI \" Person", "foiperson@localhost", "\"FOI \\\" Person\" <foiperson@localhost>"]
            got: ["FOI \\\" Person", "foiperson@localhost", "\"FOI \\\\\\\" Person\" <foiperson@localhost>"]

       (compared using ==)
     # ./spec/lib/mail_handler/mail_handler_spec.rb:219:in `should_render_from_address'
     # ./spec/lib/mail_handler/mail_handler_spec.rb:259:in `block (2 levels) in <top (required)>'

Finished in 12 minutes 39 seconds (files took 1 minute 32.65 seconds to load)
2962 examples, 3 failures, 1 pending

Failed examples:

rspec ./spec/helpers/public_body_helper_spec.rb:123 # PublicBodyHelper#type_of_authority when in a non-default locale creates the anchor href in the correct locale
rspec ./spec/controllers/general_controller_spec.rb:402 # GeneralController when using xapian search when passed a non-HTML request treats invalid formats as html
rspec ./spec/lib/mail_handler/mail_handler_spec.rb:258 # when deriving a name, email and formatted address from a message from a line should quote a name with quotes in it
```

File: app/controllers/admin_public_body_controller.rb [hunk 1]
```diff
@@ -242,11 +243,33 @@ def lookup_query
...
+               LIKE lower('%'||?||'%' ))
+               AND (public_body_translations.locale = '#{underscore_locale}')
+              EOF
+
+              [query_str, @query, @query, @query]
+            else
+              <<-EOF.strip_heredoc
+              public_body_translations.locale = '#{underscore_locale}'
+              EOF
+            end
```
Comment (lizconlan) [inline app/controllers/admin_public_body_controller.rb]: yay, this is so much clearer 🏆 🍰

File: spec/controllers/user_controller_spec.rb [hunk 2]
```diff
@@ -665,7 +665,7 @@ def do_signin(email, password)
     post :signup, { :user_signup => { :email => 'in-use@localhost', :name => 'Mr Suspected-Hacker',
                                       :password => 'sillypassword', :password_confirmation => 'mistyped' }
                     }
-    expect(assigns[:user_signup].errors[:password]).to eq(['Please enter the same password twice'])
+    expect(assigns[:user_signup].errors[:password_confirmation]).to eq(['Please enter the same password twice'])
```
Comment (garethrees) [inline spec/controllers/user_controller_spec.rb]: Fixed in f225f40

**Comment by garethrees** [comment]
Closing for a new PR

**Pull request closed without merging**